src/components/analytics/EndOfDay.vue

```diff
@@ -0,0 +1,204 @@
+        :disabled="!admin"
+      >
+        <option
+          class="batches-op"
+          v-for="company in companies"
+          v-bind:key="company.id"
+          v-bind:value="company.id"
+        >
+          {{ company.company_name }}
+        </option>
+      </select>
+    </div>
+    <table class="tble-mt">
+      <colgroup>
+        <col span="1" style="width: 25%" />
+        <col span="1" style="width: 25%" />
+        <col span="1" style="width: 25%" />
+        <col span="1" style="width: 25%" />
+      </colgroup>
+      <thead>
+        <tr class="fr-row header">
+          <th scope="col">User Name</th>
+          <th scope="col">User Type</th>
+          <th scope="col">Total Orders</th>
+          <th scope="col">Total Amount</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="data in salesanalytics.result" :key="data">
+          <td>{{ data.username }}</td>
+          <td>{{ data.user_type }}</td>
+          <td>{{ data.order_count }}</td>
+          <td>{{ data.total_amount }}</td>
+        </tr>
+      </tbody>
+    </table>
+    <div class="d-flex justify-content-between flex-row">
+      <div class="p-2"><span> Total Amount: </span>
+        <span>{{ salesanalytics.total_amount }}</span></div>
+      <div class="p-2"><span> Total Quantity: </span>
+        <span>{{ salesanalytics.total_quantity }}</span></div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import { mapActions, mapGetters } from "vuex";
+import { ActionTypes } from "@/store/modules/order/actions";
+import { ActionTypes as AuthActionTypes } from "@/store/modules/auth/actions";
+
+export default defineComponent({
+  name: "EndOfDay",
+  components: {},
+  data() {
+    const date = new Date();
+    const dateStr = `${date.getFullYear()}-${
+      date.getMonth() + 1
+    }-${date.getDate()}`;
+    return {
+      startDate: dateStr,
+      endDate: dateStr,
+      company: 0,
+    };
+  },
+  computed: {
+    ...mapGetters({
+      salesanalytics: "getSalesanalytics",
+      companies: "getInventoryCompanies",
+      userdata: "getUser",
+    }),
+    admin() {
+      const allowedRoles = ["SUPER_ADMIN", "ADMIN"];
+      const allowedCompanies = ["PARENT", "STORE"];
+      if (
+        this.userdata != null &&
+        allowedRoles.includes(this.userdata.user_type) &&
+        allowedCompanies.includes(this.userdata.company.company_type)
+      ) {
+        return true;
+      }
+      return false;
+    },
+    dateValidation: function (): string | null {
+      if (
+        this.startDate !== undefined &&
+        this.endDate !== undefined &&
+        this.startDate !== "" &&
+        this.endDate !== "" &&
+        Date.parse(this.startDate) <= Date.parse(this.endDate)
+      ) {
+        return null;
+      }
+      return "invalid date range";
+    },
+  },
+  methods: {
+    ...mapActions({
+      fetchSalesanalytics: ActionTypes.FETCH_SALESANALYTICS,
```

Comment:
where have you updated analytics model for new parameters of total_count and total_amount ?